cmd/downloader/downloader.go

```diff
@@ -218,74 +105,57 @@ func ExtractZipArchive(archiveFile, extractPath string, service Service) {
 			outfile.Close()
 		}
 	}
+	return nil
 }
 
-func ExtractTarGzipArchive(archiveFile, extractPath string, service Service) {
+// ExtractTarGzipArchive processes a tarball file and extracts a
+// single file from the service definition.
+func ExtractTarGzipArchive(archiveFile, extractPath string, service types.Service) error {
+	var outputPath string = ""
 	file, _ := os.Open(archiveFile)
 	archive, err := gzip.NewReader(file)
-	CheckError(err)
+	utils.CheckError(err)
```

Comment:
If we're now following Go best practices and returning an `error` instead of just panicing all the time, we might as well do it for all errors, yeah?